clients/python/lorax/client.py

```diff
@@ -64,7 +65,72 @@ def __init__(
         self.headers = headers
         self.cookies = cookies
         self.timeout = timeout
-        self.session = session or requests.Session()
+        self.session = session
+
+    def _create_session(self):
+        """
+        Create a new session object to make HTTP calls.
+        """
+        self.session = requests.Session()
+
+        retry_strategy = Retry(
+            total=5,
+            backoff_factor=2,
+            status_forcelist=[104, 429, 500, 502, 503, 504],
+        )
+        adapter = HTTPAdapter(
+            max_retries=retry_strategy,
+        )
+        self.session.mount("https://", adapter)
+        self.session.mount("http://", adapter)
+    
+    def _post(self, json: dict, stream: bool = False) -> requests.Response:
+        """
+        Given inputs, make an HTTP POST call
+
+        Args:
+            json (`dict`):
+                HTTP POST request JSON body
+
+            stream (`bool`):
+                Whether to stream the HTTP response or not
+        
+        Returns: 
+            requests.Response: HTTP response object
+        """
+        # Instantiate session if currently None
+        if not self.session:
+            self._create_session()
+
+        # Retry upto 2 times with a fresh session for Connection Errors
+        total_retries = 2
```

Comment:
Can you make this configurable in the client initializer?

Reply:
Done ✅